src/rp2_common/pico_cyw43_arch/cyw43_arch.c

```diff
@@ -123,6 +123,18 @@ void cyw43_hal_get_mac(__unused int idx, uint8_t buf[6]) {
     memcpy(buf, cyw43_state.mac, 6);
 }
 
+// e.g. 112233445566
+// chr_off: index of character in mac to start
+// chr_len: length of result
```

Comment:
Is it possible for a bad combination of `chr_off` and `chr_len` values to cause you to "fall off the end" of the mac, before reading the full `chr_len`? Perhaps it's worth adding an assert to guard against that?

Reply:
ok - I've had a go at fixing that.